contrib/linux-kernel/zstd_compress_module.c

```diff
@@ -133,7 +133,7 @@ EXPORT_SYMBOL(zstd_init_cstream);
 size_t zstd_reset_cstream(zstd_cstream *cstream,
 	unsigned long long pledged_src_size)
 {
-	return ZSTD_resetCStream(cstream, pledged_src_size);
+	return ZSTD_CCtx_reset(cstream, pledged_src_size);
```

Comment:
This is incorrect, this is the definition of `ZSTD_resetCStream()`. You should do the same here.

https://github.com/facebook/zstd/blob/930b5cecaa033c3aa6017b246bb804ef576942b4/lib/compress/zstd_compress.c#L5204-L5212

E.g. 

```c
if (pledged_src_size == 0)
    pledged_src_size = ZSTD_CONTENTSIZE_UNKNOWN;
ZSTD_FORWARD_IF_ERR( ZSTD_CCtx_reset(zstream, ZSTD_reset_session_only) ); 
ZSTD_FORWARD_IF_ERR( ZSTD_CCtx_setPledgedSrcSize(zstream, pledged_src_size) ); 
```

Since there are currently no users that depend on that behavior. But then you'll also

Reply:
How about changing cstream to zcs and pledged_src_size to pledgedSrcSize in contrib/linux-kernel/zstd_compress_module.c? This would be a better implementation than to modify the core function of ZSTD_CCtx_reset which might break the functionality of it in other use cases.

Reply:
Update: I just tested with ZSTD_CCtx_setPledgedSrcSize and it seemed to work fine(except I had to change pledged_src_size to pledgedSrcSize in zstd_compress_module.c).

I'll push a patch with the change I stated.

cc: @terrelln

Reply:
@cyberknight777 I'm not asking you to change `ZSTD_CCtx_reset()`, I'm asking you to make `zstd_reset_cstream()` functionally identical to `ZSTD_resetCStream()`.

You still need to call `ZSTD_CCtx_reset()`. Also, please keep the name `pledged_src_size`.

The function should be (with tab indentation):

```c
size_t zstd_reset_cstream(zstd_cstream *cstream,
	unsigned long long pledged_src_size)
{
    if (pledged_src_size == 0)
        pledged_src_size = ZSTD_CONTENTSIZE_UNKNOWN;
    ZSTD_FORWARD_IF_ERR( ZSTD_CCtx_reset(cstream, ZSTD_reset_session_only) ); 
    ZSTD_FORWARD_IF_ERR( ZSTD_CCtx_setPledgedSrcSize(cstream, pledged_src_size) );
    return 0;
}
```

Reply:
Alright, I see what you mean. I'm pushing another patch in a few.

Reply:
@terrelln any comments on the patch I pushed?